addons/mail/controllers/discuss/channel.py

```diff
@@ -141,3 +141,33 @@ def discuss_channel_fold(self, channel_id, state, state_count):
         if not member:
             raise NotFound()
         return member._channel_fold(state, state_count)
+
+    @http.route("/discuss/channel/join", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
+    def discuss_channel_join(self, channel_id):
+        channel = request.env["discuss.channel"].search([("id", "=", channel_id)])
+        if not channel:
+            raise NotFound()
+        channel._find_or_create_member_for_self()
+        return Store(channel).get_result()
+
+    @http.route("/discuss/channel/sub_channel/create", methods=["POST"], type="json", auth="public")
+    def discuss_channel_sub_channel_create(self, parent_channel_id, from_message_id=None, name=None):
+        channel = request.env["discuss.channel"].search([("id", "=", parent_channel_id)])
+        if not channel:
+            raise NotFound()
+        return channel._create_sub_channel(from_message_id, name)
```

Comment:
This crashes for guest (just creating a sub channel) we should probably use sudo there if there are no specific restrictions we should put in place other than main channel access.

Maybe validate channel type? As currently we can create a sub channel for any type of thread by called the RPC, but not from UI. Why the restriction actually?

Anyway people with rights (internal users) can create them with ORM too. Do we need constraints?

Reply:
I added a constraint to restrict sub-channel creation to channel_type channel.